Update community-plugins.json

shmolf opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
This looks a bit simple, and could have also been a CSS snippet.

Including the main plugin class in a file named `PluginSettings` feels a bit odd.

[app.workspace.containerEl;](https://github.com/shmolf/obsidian-editor-background/blob/0b7b0b299ed593f2cbdcea6145b16e47732ab987/src/PluginSettings.ts#L39)
It's highly discouraged that you use the global app instance in your plugin. It's currently accessible from the global context for debugging purposes, and might be removed in the future. Instead, you should use the reference to App that you have from your plugin instance.

This plugin won't work in pop out windows.
See here how to update your plugin to support pop out windows:
<https://forum.obsidian.md/t/plugin-developers-how-to-update-your-plugin-to-support-pop-out-windows/40228>

shmolf commented:
I've been trying out different things to support the popout window, but I'm having trouble access the `DOM` of the popout window, to add the CSS variables to an element of the tree.

Things I've tried:
```ts
    this.app.workspace.onLayoutReady(() => this.UpdateBackground({ fromWhere: 'layout-ready' }));
    this.app.workspace.on('layout-change', () => this.UpdateBackground({ fromWhere: 'layout-change' }));
    this.app.workspace.on('window-open', (_, window) => {
      console.log(activeWindow.document == activeDocument);
      console.log(activeDocument);
      console.log(activeWindow.document);
      console.log(window.document);
      this.UpdateBackground({ document: activeWindow.document, fromWhere: 'window-open 1' });
      this.UpdateBackground({ document: window.document, fromWhere: 'window-open 2' });
      this.UpdateBackground({ fromWhere: 'window-open 3' });
    });
```

```js
  UpdateBackground({
    document = activeDocument
  }: { document?: Document, fromWhere?: string } = {}): void {
    // const container = this.app.workspace.containerEl;
    const containers = document.querySelectorAll('.cm-editor.ͼ1.ͼ2.ͼq') as NodeListOf<HTMLElement>;

    containers.forEach((container) => {
      container.style.setProperty('--obsidian-editor-background-image', `url('${this.settings.imageUrl}')`);
      container.style.setProperty('--obsidian-editor-background-opacity', `${this.settings.opacity}`);
      container.style.setProperty('--obsidian-editor-background-bluriness', `blur(${this.settings.bluriness})`);
      container.style.setProperty('--obsidian-editor-background-input-contrast', this.settings.inputContrast ? '#ffffff17' : 'none');
    });
  }
```

shmolf commented:
@joethei Ignore my prior comment. I found a solution, and have pushed up the changes, and a new release.

shmolf commented:
@liamcain @joethei Is there anything else needed of this plugin, before it can be considered ready to be merged?

Or, if there's no expectation that it'll be merged, then I'd rather close this PR, rather than handle the merge conflicts each time they occur.

joethei commented:
You don't need to fix the merge conflicts everytime, we will do that once the plugin has been accepted.

liamcain commented:
Try this:

```ts
import { Plugin } from 'obsidian';
import { UrlSettingsTab } from './PluginSettingsTab';

interface PluginSettings {
  imageUrl: string;
  opacity: number;
  bluriness: string;
  inputContrast: boolean;
}

export const DEFAULT_SETTINGS: Partial<PluginSettings> = {
  imageUrl: 'protocol://domain.tld/path/to/image.png',
  opacity: 0.3,
  bluriness: 'low',
  inputContrast: false,
};

export default class BackgroundPlugin extends Plugin {
  settings: PluginSettings;

  async onload() {
    await this.loadSettings();

    this.addSettingTab(new UrlSettingsTab(this.app, this));
    this.app.workspace.onLayoutReady(() => this.UpdateBackground(document));
    this.app.workspace.on('window-open', (win) => this.UpdateBackground(win.document));
  }

  async loadSettings() {
    this.settings = Object.assign({}, DEFAULT_SETTINGS, await this.loadData());
  }

  async saveSettings() {
    await this.saveData(this.settings);
    this.UpdateBackground();
  }

  UpdateBackground(doc: Document = activeDocument) {
    document.body.style.setProperty('--obsidian-editor-background-image', `url('${this.settings.imageUrl}')`);
    document.body.style.setProperty('--obsidian-editor-background-opacity', `${this.settings.opacity}`);
    document.body.style.setProperty('--obsidian-editor-background-bluriness', `blur(${this.settings.bluriness})`);
    document.body.style.setProperty('--obsidian-editor-background-input-contrast', this.settings.inputContrast ? '#ffffff17' : 'none');
  }
}
```

liamcain commented:
I don't think you should try to keep the css vars on the editor. It's easier to just add it to the body of each window once.

shmolf commented:
> I don't think you should try to keep the css vars on the editor. It's easier to just add it to the body of each window once.

That makes sense!

@liamcain  I've created a new version, using the solution you provided. Thank you!

shmolf commented:
Wanted to check back on this PR. See if there was anything else needed

shmolf commented:
Are there any more changes expected of this?

The last [update to the plugin](https://github.com/shmolf/obsidian-editor-background/pull/10) hopefully addresses the concerns from above.

liamcain commented:
Hi, I'm really sorry for such a long delay in getting back to this plugin. I somehow missed the notifications in my review queue 😢

Just took another look. The changes look great! I appreciate all your patience.
